From: Lasse Collin <lasse.collin@tukaani.org>
Date: Thu, 3 Apr 2025 14:34:30 +0300
Subject: liblzma: Add assertions to lzma_bufcpy()

(cherry picked from commit b9d168eee4fb6393b4fe207c0aeb5faee316ca1a)
---
 src/liblzma/common/common.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/liblzma/common/common.c b/src/liblzma/common/common.c
index a708fdf..d43e4d6 100644
--- a/src/liblzma/common/common.c
+++ b/src/liblzma/common/common.c
@@ -95,6 +95,12 @@ lzma_bufcpy(const uint8_t *restrict in, size_t *restrict in_pos,
 		size_t in_size, uint8_t *restrict out,
 		size_t *restrict out_pos, size_t out_size)
 {
+	assert(in != NULL || *in_pos == in_size);
+	assert(out != NULL || *out_pos == out_size);
+
+	assert(*in_pos <= in_size);
+	assert(*out_pos <= out_size);
+
 	const size_t in_avail = in_size - *in_pos;
 	const size_t out_avail = out_size - *out_pos;
 	const size_t copy_size = my_min(in_avail, out_avail);
